charts/k8ssandra/templates/cassandra/cassdc.yaml

```diff
@@ -1,7 +1,13 @@
 {{- $datacenter := (index .Values.cassandra.datacenters 0) -}}
 {{- $medusaImage := (printf "%s:%s" .Values.medusa.image.repository .Values.medusa.image.tag) -}}
-{{- $defaultSecurityCtx := (include "k8ssandra-common.container.security-ctx-defaults" .) -}}
-{{- $defaultPodSecurityCtx := (include "k8ssandra-common.pod.pod-security-ctx-defaults" .) -}}
+{{- $cassServerCfgSecurityCtx := (include "k8ssandra-common.container.cass-svrcfg-security-ctx-defaults" .) -}}
```

Comment:
It looks like there is a separate template used for each container. What is the value in using the templates vs just making sure the security context is configurable for each container and providing a default value? A template would make more sense to me if it was shared and reused, but that doesn't appear to be the case.

Reply:
Default security settings between containers are currently setup to be used as a 1:1 mapping for the k8ssandra common templates.  The `SecurityContext` default settings applied by K8ssandra _may_ have cases in the future where the templates are shared among container settings.  In this PR they are independent of one another to allow for flexibility of change as we evolve the overall set of default security properties. 

Having these scoped to owned containers (and pods) made it easier to adjust a security policy without impacting other components.  Reusing across unrelated components became time consuming to test and track how a new setting impacted the others.  Being a security related feature and its importance, redundancy & clarity seemed more appropriate for these.

One convenient side-effect of the template usage is when a user has `SecurityContext` properties existing in the `values.yaml`, it becomes an easy way to identify that the K8ssandra default security policy is being overridden for a particular container and/or pod.  

Also, not having defaults existing directly in the `values.yaml` is less verbose prior to template rendering and reduces the chance for misconfigured applied security settings.  Our intent I believe is that it will be rare when needing to override the K8ssandra security defaults as they adhere to k8s best-practices.

Reply:
Moving forward with changes to remove from templates and apply to directly to the `values.yaml` for the configuration by users.